src/settings.rs

```diff
@@ -0,0 +1,20 @@
+use glib::variant::FromVariant;
+use glib::{IsA, ToVariant};
+use {Settings, SettingsExt};
+
+pub trait SettingsExtManual {
+    fn get<U: FromVariant>(&self, key: &str) -> Option<U>;
+
+    fn set<U: ToVariant>(&self, key: &str, value: &U) -> bool;
+}
+
+impl<O: IsA<Settings>> SettingsExtManual for O {
+    fn get<U: FromVariant>(&self, key: &str) -> Option<U> {
+        self.get_value(key)
+            .and_then(|val| FromVariant::from_variant(&val))
+    }
+
+    fn set<U: ToVariant>(&self, key: &str, value: &U) -> bool {
+        self.set_value(key, &value.to_variant())
```

Comment:
This `bool` should probably be a `Result<(), glib::BoolError>` :) Probably also for any other `set()` functions on `Settings`

Reply:
Hm, if you think that's in the scope of this PR, sure. Should I make wrapper functions for set_* or move the entirety of auto/settings.rs's set_* into this?

Reply:
You can tell the code generator, see https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/blob/master/Gir_Gst.toml#L1291

It's an easy change so I thought it would be good to have as part of this PR

Reply:
Oh, alright, hadn't known about that. Thanks for the pointer :)